src/Stratis.Bitcoin.Features.Dns/WhitelistManager.cs

```diff
@@ -63,26 +69,22 @@ public class WhitelistManager : IWhitelistManager
         /// <param name="loggerFactory">The factory to create the logger.</param>
         /// <param name="peerAddressManager">The manager implementation for peer addresses.</param>
         /// <param name="dnsServer">The DNS server.</param>
-        /// <param name="nodeSettings">The node settings.</param>
-        public WhitelistManager(IDateTimeProvider dateTimeProvider, ILoggerFactory loggerFactory, IPeerAddressManager peerAddressManager, IDnsServer dnsServer, NodeSettings nodeSettings, DnsSettings dnsSettings)
+        /// <param name="connectionSettings">The connection settings.</param>
+        public WhitelistManager(IDateTimeProvider dateTimeProvider, ILoggerFactory loggerFactory, IPeerAddressManager peerAddressManager, IDnsServer dnsServer, ConnectionManagerSettings connectionSettings, DnsSettings dnsSettings)
         {
             Guard.NotNull(dateTimeProvider, nameof(dateTimeProvider));
             Guard.NotNull(loggerFactory, nameof(loggerFactory));
             Guard.NotNull(peerAddressManager, nameof(peerAddressManager));
             Guard.NotNull(dnsServer, nameof(dnsServer));
-            Guard.NotNull(nodeSettings, nameof(nodeSettings));
             Guard.NotNull(dnsSettings, nameof(dnsSettings));
-            Guard.NotNull(dnsSettings.DnsHostName, nameof(dnsSettings.DnsHostName));
-            Guard.NotNull(nodeSettings.ConnectionManager, nameof(nodeSettings.ConnectionManager));
+            Guard.NotNull(connectionSettings, nameof(connectionSettings));
 
             this.dateTimeProvider = dateTimeProvider;
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
             this.peerAddressManager = peerAddressManager;
             this.dnsServer = dnsServer;
-            this.dnsPeerBlacklistThresholdInSeconds = dnsSettings.DnsPeerBlacklistThresholdInSeconds;
-            this.dnsHostName = dnsSettings.DnsHostName;
-            this.externalEndpoint = nodeSettings.ConnectionManager.ExternalEndpoint;
-            this.fullNodeMode = dnsSettings.DnsFullNode;
+            this.dnsSettings = dnsSettings;                      
```

Comment:
Yes, this is what I was getting at in the previous PRs! Good job.

Reply:
Yes, I saw the light. Thank you.